core/src/Streamly/Internal/FileSystem/PosixPath.hs

```diff
@@ -234,16 +258,21 @@ For APIs that have not been released yet.
 -- Path components may have limits.
 -- Total path length may have a limit.
 
+#if defined(IS_PORTABLE)
+type OS_PATH_TYPE = OS_PATH
+type OS_WORD_TYPE = OS_WORD
+type OS_CSTRING_TYPE = OS_CSTRING
+#else
 -- | A type representing file system paths on OS_NAME.
 --
--- A OS_PATH is validated before construction unless unsafe constructors are
+-- A OS_PATH_TYPE is validated before construction unless unsafe constructors are
 -- used to create it. For validations performed by the safe construction
 -- methods see the 'fromChars' function.
 --
 -- Note that in some cases the file system may perform unicode normalization on
 -- paths (e.g. Apple HFS), it may cause surprising results as the path used by
 -- the user may not have the same bytes as later returned by the file system.
-newtype OS_PATH = OS_PATH (Array WORD_TYPE)
+newtype OS_PATH = OS_PATH (Array OS_WORD_TYPE)
```

Comment:
Do we define a newtype in the `Path` module? The portable module should not have this right?

Reply:
Oh nvm. This is under `else` branch.